### ISSUE TITLE
Hyperlinks not working

### ISSUE THREAD

**Issue by mac-f** [mozilla/pdf.js#9637]
Attach (recommended) or Link to PDF file here:

[1260415x.pdf](https://github.com/mozilla/pdf.js/files/1881234/1260415x.pdf)

Configuration:
- Web browser and its version: Microsoft Edge 41.16299.248.0 / Firefox 59.0.2 / Google Chrome 65.0.3325.181
- Operating system and its version: Windows 10.0.16299
- PDF.js version: latest I suppose, Firefiox console reports PDF.js 2.0.288
- Is a browser extension: buit-in apparently

Steps to reproduce the problem:
1. Open the attached file
2. Try clicking the hyperlinks (on the first page the text "67-400", on the second page the text "67-420")

What is the expected behavior? (add screenshot)

The first hyperlink should open file ""../../../ra/67/00/1260400x.pdf"", the second one should open "../../../ra/67/00/1260420x.pdf".
By the way, the attached file 1260415x.pdf is also in the same folder (/Datas/War/EPC/War_00/Typ126/Case1/ra/67/00 is the complete path).
In 'normal' pdf viewers (Acrobat DC, Foxit, PDF-XChange) it does function as expected.

What went wrong? (add screenshot)

The hyperlinked files are not opened / shown.
(On hover in Edge and Chrome, the mouse cursor is correctly changed to "pointer", if Firefox it is incorrectly the "text" cursor; on click in Edge, the hyperlink text is surrounded by dotted lines, in Chrome and Firefox nothing appears to happen.)
All other files (several hundreds of them) belonging to the same set of documents behave in the same way; hyperlinks in the documents do not work when the documents are opened in a webbrowser, in a 'normal' pdf viewers they do.

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

I have also tried it in https://mozilla.github.io/pdf.js/web/viewer.html, it does not function as expected there either.
https://brendandahl.github.io/pdf.js.utils/browser/ does show the correct pathnames in Trailer -> Root -> Pages -> Kids -> 0 -> Annots -> 0 -> A -> F and Trailer -> Root -> Pages -> Kids -> 1 -> Annots -> 0 -> A -> F

**Comment by Snuffleupagus** [comment]
> The first hyperlink should open file ""../../../ra/67/00/1260400x.pdf"", the second one should open "../../../ra/67/00/1260420x.pdf".

First of all, please note that using relative links in PDF files is never a good idea, since it means that the files aren't actually as portable as they ought be.

However, since relative links are unfortunately somewhat common, in the Firefox built-in version of PDF.js we'll attempt to create valid absolute links by using the current document URL in to improve the UX; please refer to PR #7689.

> (/Datas/War/EPC/War_00/Typ126/Case1/ra/67/00 is the complete path)

The one caveat here is that browsers will, for security reasons, not allow linking to `file://` URLs which should explain why this unfortunately cannot work for *local* files (which I'm assuming is what the problem is about).

To summarize: the built-in PDF viewer in Firefox *will* attempt to fix relative links in PDF files loaded over the internet (e.g. with `http://`, `https://`, ... protocols). However this unfortunately cannot be supported when opening local (i.e. `file://`) files, and this issue is therefore invalid.
